src/main/java/com/ita/if103java/ims/service/impl/ItemServiceImpl.java

```diff
@@ -67,75 +75,113 @@ public ItemServiceImpl(ItemDtoMapper itemDtoMapper, SavedItemDtoMapper savedItem
+                user.getUser().getAccountId(),
+                itemTransaction.getDestinationWarehouseId(), user.getUser().getId(), EventName.LOW_SPACE_IN_WAREHOUSE
+                , null));
+            throw new ItemNotEnoughCapacityInWarehouseException("Can't add savedItemDto in warehouse because it " +
+                "doesn't  " +
+                "have enough capacity {warehouse_id = " + itemTransaction.getDestinationWarehouseId() + "}");
         }
-        throw new SavedItemNotFoundException("Failed to get savedItem during `create` {account_id = " + itemTransaction.getItemDto().getAccountId() + "}");
+
+
     }
 
-    private boolean isEnoughCapacityInWarehouse(ItemTransactionRequestDto itemTransaction, Function<ItemTransactionRequestDto, Long> function) {
-        float volume = getVolumeOfPassSavedItems(itemTransaction) + itemTransaction.getQuantity() * itemTransaction.getItemDto().getVolume();
+
+    private boolean isEnoughCapacityInWarehouse(ItemTransactionRequestDto itemTransaction, Long accountId) {
+        float volume =
+            getVolumeOfPassSavedItems(itemTransaction, accountId) + itemTransaction.getQuantity() * itemTransaction.getItemDto().getVolume();
         return warehouseDao.findById(itemTransaction.getDestinationWarehouseId()).getCapacity() >= volume;
     }
 
-    private float getVolumeOfPassSavedItems(ItemTransactionRequestDto itemTransaction) {
+    private float getVolumeOfPassSavedItems(ItemTransactionRequestDto itemTransaction, Long accountId) {
         float volumePassSavedItems = 0;
         Long warehouseId = itemTransaction.getDestinationWarehouseId();
 
         if (savedItemDao.existSavedItemByWarehouseId(warehouseId)) {
             for (SavedItem savedItem : savedItemDao.findSavedItemByWarehouseId(warehouseId)) {
-                Item item = itemDao.findItemById(savedItem.getItemId());
+                Item item = itemDao.findItemById(savedItem.getItemId(), accountId);
                 volumePassSavedItems += savedItem.getQuantity() * item.getVolume();
-                System.out.println(volumePassSavedItems + item.getName() + item.getVolume() + savedItem.getQuantity());
             }
         }
-        System.out.println(volumePassSavedItems);
         return volumePassSavedItems;
     }
 
 
-    private boolean isLowSpaceInWarehouse(ItemTransactionRequestDto itemTransaction) {
-        if (getVolumeOfPassSavedItems(itemTransaction) == 0) {
+    private boolean isLowSpaceInWarehouse(ItemTransactionRequestDto itemTransaction, Long accountId) {
+        float volume = getVolumeOfPassSavedItems(itemTransaction, accountId);
+        if (volume == 0) {
             return true;
         } else {
-            return getVolumeOfPassSavedItems(itemTransaction) * 100 / warehouseDao.findById(itemTransaction.getDestinationWarehouseId()).getCapacity() < 90.0;
+            return volume * 100 / warehouseDao.findById(itemTransaction.getDestinationWarehouseId()).getCapacity() < 90.0;
```

Comment:
here I would change < to > in order to corresponds to the method name and get rid of inversion in the place where this method is called